src/main/java/core/basesyntax/model/User.java

```diff
@@ -1,13 +1,22 @@
 package core.basesyntax.model;
 
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class User {
+    private static AtomicLong idCounter = new AtomicLong(1);
     private Long id;
     private String login;
     private String password;
     private Integer age;
 
+    public User(String login, String password, Integer age) {
+        this.id = idCounter.get();
```

Comment:
The `id` is assigned using `idCounter.get()`, which does not increment the counter. This means all users will have the same `id`. Consider using `idCounter.getAndIncrement()` to ensure each user gets a unique `id`.